src/psbt.c

```diff
@@ -0,0 +1,2120 @@
+#include "internal.h"
+
+#include "ccan/ccan/base64/base64.h"
+#include "ccan/ccan/build_assert/build_assert.h"
+
+#include <include/wally_crypto.h>
+#include <include/wally_transaction.h>
+#include <include/wally_psbt.h>
+
+#include <limits.h>
+#include <stdbool.h>
+#include "transaction_int.h"
+#include "script_int.h"
+
+#define TX_CHECK_OUTPUT if (!output) return WALLY_EINVAL; else *output = NULL
+#define TX_OUTPUT_ALLOC(typ) \
+    *output = wally_malloc(sizeof(typ)); \
+    if (!*output) return WALLY_ENOMEM; \
+    wally_clear((void *)*output, sizeof(typ)); \
+    result = (typ *) *output;
+
+static bool clone_bytes(unsigned char **dst, const unsigned char *src, size_t len)
+{
+    if (!len)
+        return true;
+    *dst = wally_malloc(len);
+    if (*dst)
+        memcpy(*dst, src, len);
+    return *dst != NULL;
+}
```

Comment:
Would it be possible to move this function into maybe internal.c and reuse it from psbt.c and transaction.c?

Reply:
Should I also move other things that are shared like `TX_CHECK_OUTPUT`, `analyze_tx`, etc or should those go in a more transaction specific internal thing?

Reply:
I think ideally any common code would move to shared files. Whether that's internal.c/h or a transaction specific file I am fairly relaxed.

Reply:
Done